src/test/pages/AdminUserManager.test.tsx

```diff
@@ -0,0 +1,446 @@
+        const roleOptions = screen.getByTestId("modify-role-options");
+        const buttons = Array.from(roleOptions.querySelectorAll("button"));
+        buttons.forEach((button, index) => {
+          expect(button).toHaveTextContent(expectedRoles[index]);
+        });
+        expect(within(roleOptions).queryByText("ADMIN", { selector: "button" })).toBeNull();
+        if (role === "EVENT_MANAGER") {
+          ["TUTOR", "EVENT_MANAGER"].forEach((button) =>
+            expect(within(roleOptions).queryByText(button, { selector: "button" })).toBeNull(),
+          );
+        }
+      },
+    );
+
+    it("does not show Email Status button for EVENT_MANAGER user", async () => {
+      await renderUserManager({ role: "EVENT_MANAGER" });
+      const { emailStatus } = adminButtons;
+      expect(emailStatus()).not.toBeInTheDocument();
+    });
+
+    it("shows correct options for modifying email status for ADMIN user", async () => {
+      await renderUserManager();
+      const statusOptions = screen.getByTestId("email-status-options");
+      const buttons = Array.from(statusOptions.querySelectorAll("button"));
+      const expectedOptions = ["NOT_VERIFIED", "DELIVERY_FAILED"];
+      buttons.forEach((button, index) => {
+        expect(button).toHaveTextContent(expectedOptions[index]);
+      });
+    });
+
+    it("shows a list of users with expected columns and data after a search is done", async () => {
+      await renderUserManager();
+      const { resultsTable, tableRows, firstUserDetails } = await searchAndWaitForResults();
+      const headers = within(resultsTable).getAllByRole("columnheader");
+      const expectedHeaders = [
+        "Select",
+        "Actions",
+        "Name",
+        "Email",
+        "User role",
+        "School",
+        "Verification status",
+        "Teacher pending?",
+        "Member since",
+        "Last seen",
+      ];
+      headers.forEach((header, index) => expect(header).toHaveTextContent(expectedHeaders[index]));
+      expect(tableRows).toHaveLength(4);
+      const expectedUserData = [
+        "Admin, Test",
+        "test-admin@test.com",
+        "ADMIN",
+        "Test School",
+        "VERIFIED",
+        "N",
+        FRIENDLY_DATE_AND_TIME.format(mockUser.registrationDate),
+        FRIENDLY_DATE_AND_TIME.format(mockUser.lastSeen),
+      ];
+      firstUserDetails.slice(2).forEach((cell, index) => expect(cell).toHaveTextContent(expectedUserData[index]));
+    });
+
+    it("shows a checkbox for selecting a user and buttons to View, Edit, Delete or Reset password", async () => {
+      await renderUserManager();
+      const { firstUserDetails } = await searchAndWaitForResults();
+      const checkbox = within(firstUserDetails[0]).getByRole("checkbox");
+      expect(checkbox).toBeInTheDocument();
+      expect(checkbox).not.toBeChecked();
+      const expectedButtons = ["View", "Edit", "Delete", "Reset password"];
+      const buttons = Array.from(firstUserDetails[1].children);
```

Comment:
You should probably assert the expected length of `buttons`: if this ever returns an empty array, the assertions in the next line won't run.